bokeh/util/serialization.py

```diff
@@ -150,6 +150,10 @@ def transform_array_to_list(array):
         transformed[np.isposinf(array)] = 'Infinity'
         transformed[np.isneginf(array)] = '-Infinity'
         return transformed.tolist()
+    elif (array.dtype.kind == 'O' and pd.isnull(array).any()):
```

Comment:
we don't assume `pandas` as a hard dependency, `pd` can be none if `optional_import` above fails, so will need to check that.